pkg/registry/rbac/rest/storage_rbac.go

```diff
@@ -206,7 +215,8 @@ func (p *PolicyData) EnsureRBACPolicy() genericapiserver.PostStartHookFunc {
 					Client:  reconciliation.ClusterRoleModifier{Client: clientset.ClusterRoles()},
 					Confirm: true,
 				}
-				err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				// ServiceUnavailble error is returned when the API server is blocked by storage version updates
+				err := retryOnConflictOrServiceUnavailable(retry.DefaultBackoff, func() error {
```

Comment:
An alternative to this is to permit self-requests even when blocking other requests, right? Did we think about that?

Reply:
Another option is to do a GET first. This might be useful anyway, as 99.99% of the time this stuff will already be correct.

Reply:
> An alternative to this is to permit self-requests even when blocking other requests

The purpose of the filter is to make sure no write request gets through before the apiserver updates the corresponding storage version record. I don't see how self-requests are different from other requests.  

> Another option is to do a GET first

Do you mean get the rule or get the storage version?

Reply:
Getting the rule.

Reply:
The ReconcileRole logic does do a [get](https://github.com/kubernetes/kubernetes/blob/bd95fb101dd83cfca9d95e55b2f0d29720016e51/staging/src/k8s.io/component-helpers/auth/rbac/reconciliation/reconcile_role.go#L110) first and skips if no Reconciliation is needed

Reply:
Great.

Reply:
I guess in the future we might want to think about just retrying, period, rather than build up a long list of reasons.